warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EventDataQueryExpressionVisitor.java

```diff
@@ -141,17 +136,77 @@ public boolean apply(Key key) {
                 }
                 
                 final String keyFieldValue = datawaveKey.getFieldValue();
-                final Set<String> normalizedFieldValues = EventDataQueryExpressionVisitor.extractNormalizedAttributes(attributeFactory, keyFieldName,
-                                keyFieldValue, key);
+                final List<Type> types = EventDataQueryExpressionVisitor.extractTypes(attributeFactory, keyFieldName, keyFieldValue, key);
+                final List<Matcher> normalizedPatternMatchers = new ArrayList<>();
+                final List<String> normalizedFieldValues = new ArrayList<>();
+                final List<LiteralRange> normalizedRanges = new ArrayList<>();
```

Comment:
Should `normalizedPatternMatchers`, `normalizedFieldValues` and `normalizedRanges` be `Set<T>` instead of a List? e.g: Do we care about duplicate values? Is deduplication handled elsewhere?

Reply:
agree